node/device/core/lib/client.js

```diff
@@ -143,7 +143,7 @@ Client.fromSharedAccessSignature = function (sharedAccessSignature, Transport) {
   if (!sharedAccessSignature) throw new ReferenceError('sharedAccessSignature is \'' + sharedAccessSignature + '\'');
 
   var sas = SharedAccessSignature.parse(sharedAccessSignature);
-  var uriSegments = sas.sr.split('/')[0];
+  var uriSegments = sas.sr.split('/');
```

Comment:
Looks like we should add a test in `node/device/core/test/_client.test.js` corresponding to this bug, if you have time feel free to do it, I'll be happy to do it if myself you don't feel like it though.